pypeit/core/tracewave.py

```diff
@@ -261,37 +296,71 @@ def trace_tilts_work(arcimg, lines_spec, lines_spat, thismask, slit_cen, inmask=
-            sub_img, tilts_guess_now, spat_order, inmask=sub_inmask, trc_inmask = tilts_sub_mask_box, fwhm=fwhm,
-            maxdev=maxdev, niter=6, idx=str(iline),show_fits=show_tracefits, xmin=0.0,xmax=float(nsub-1))
-        tilts_sub_mask_box = (extract.extract_boxcar(sub_thismask, tilts_sub_fit_out, fwhm/2.0) > 0.99*fwhm)
-        if gauss: # If gauss is set, do a Gaussian refinement to the flux weighted tracing
+
+        # Do iterative flux-weighted tracing and polynomial fitting to
+        # refine these traces. This must also be done in a loop since
+        # the sub image is different for every aperture, i.e. each
+        # aperature has its own image.
+        tilts_sub_fit_out, tilts_sub_out, tilts_sub_err_out, tilts_sub_bpm_out, tset_out \
+                = trace.fit_trace(sub_img, tilts_guess_now, spat_order,
+                                  bpm=np.invert(sub_inmask.astype(bool)),
+                                  trace_bpm=np.invert(tilts_sub_mask_box), fwhm=fwhm,
+                                  maxdev=maxdev, niter=6, idx=str(iline), debug=show_tracefits,
+                                  xmin=0.0, xmax=float(nsub-1))
+
+        # Update the spatial positions to include based on the fitted
+        # line trace positions
+        tilts_sub_mask_box = moment1d(sub_thismask, tilts_sub_fit_out, fwhm)[0] > 0.99*fwhm
+
+        # If gauss is set, do a Gaussian refinement to the
+        # flux-weighted tracing
+        if gauss:
+            # Re-check if spatial pixels should be unmasked.
             if (np.sum(tilts_sub_mask_box) < 0.8 * nsub):
                 tilts_sub_mask_box = np.ones_like(tilts_sub_mask_box)
-            tilts_sub_fit_gw, tilts_sub_gw, tilts_sub_err_gw, tset_gw = extract.iter_tracefit(
-                sub_img, tilts_sub_fit_out, spat_order, inmask=sub_inmask, trc_inmask = tilts_sub_mask_box, fwhm=fwhm,
-                maxdev=maxdev, niter=3, idx=str(iline),show_fits=show_tracefits, xmin=0.0, xmax=float(nsub-1))
-            tilts_sub_fit_out = tilts_sub_fit_gw
-            tilts_sub_out = tilts_sub_gw
-            tilts_sub_err_out = tilts_sub_err_gw
-        tilts_sub_mask_box = (extract.extract_boxcar(sub_thismask, tilts_sub_fit_out, fwhm/2.0) > 0.99*fwhm)
+            # Re-measure using Gaussian weighting and refit
+            tilts_sub_fit_out, tilts_sub_out, tilts_sub_err_out, tilts_sub_bpm_out, _ \
+                    = trace.fit_trace(sub_img, tilts_sub_fit_out, spat_order,
+                                      bpm=np.invert(sub_inmask.astype(bool)),
+                                      trace_bpm=np.invert(tilts_sub_mask_box), 
+                                      weighting='gaussian', fwhm=fwhm, maxdev=maxdev, niter=3,
+                                      idx=str(iline), debug=show_tracefits, xmin=0.0,
+                                      xmax=float(nsub-1))
+
+        # TODO: Why is this done outside of the 'if gauss' block?
```

Comment:
It looks like this is a mistake, i.e. this should be in the if gauss block. If gauss==False, this just repeats the line above.

Reply:
That's what I thought.  Moved inside the if block.